### PR TITLE
Functional Tests : BO - Orders - Shopping carts - View carts

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Functional Tests : BO - Orders - Shopping carts - View carts
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Related PRs       | N/A
| How to test?      | Run command `HEADLESS=false TEST_PATH="functional/BO/02*/05*/03*" URL_FO="https://domain.tldl/" npm run test:specific:fast-fail`
| Possible impacts? | N/A

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+      await expect(numberOfShoppingCartsAfterFilter).to.be.at.equal(1);
+
+      const textColumn = await shoppingCartsPage.getTextColumn(page, 1, 'c!lastname');
+      await expect(textColumn).to.contains('Non ordered');
+    });
+
+    it('should go the Shopping Cart details page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToShoppingCartDetailPage1', baseContext);
+
+      await shoppingCartsPage.gotoViewPage(page, 1);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]:
```suggestion
      await shoppingCartsPage.goToViewPage(page, 1);
```

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+    it('should fill the order and create it', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'fillAndCreateOrder', baseContext);
+
+      // Choose payment method
+      await addOrderPage.setPaymentMethod(page, PaymentMethods.checkPayment.displayName);
+      // Set order status
+      await addOrderPage.setOrderStatus(page, Statuses.paymentAccepted);
+      // Create the order
+      await addOrderPage.clickAndWaitForNavigation(page, addOrderPage.createOrderButton);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]: Please create a method to click on create Button on `addOrderPage`

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        page,
+        dashboardPage.ordersParentLink,
+        dashboardPage.shoppingCartsLink,
+      );
+
+      const pageTitle = await shoppingCartsPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(shoppingCartsPage.pageTitle);
+    });
+
+    it('should reset all filters and get number of shopping carts', async function () {
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]: Maybe change it to => should reset all filters and check number of shopping carts

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', 'resetFiltersFirst2', baseContext);
+
+      numberOfShoppingCarts = await shoppingCartsPage.resetAndGetNumberOfLines(page);
+      await expect(numberOfShoppingCarts).to.be.above(0);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]:
```suggestion
      const numberOfShoppingCartsAfterReset = await shoppingCartsPage.resetAndGetNumberOfLines(page);
      await expect(numberOfShoppingCartsAfterReset).to.equal(numberOfShoppingCarts);
```

File: tests/UI/pages/BO/orders/shoppingCarts/index.js [hunk 5]
```diff
@@ -327,6 +329,16 @@ class ShoppingCarts extends BOBasePage {
...
     await this.clickAndWaitForNavigation(page, sortColumnButton);
   }
+
+  /**
+   * Go to view page
+   * @param page {Page} Browser tab
+   * @param row {number} Row on table
+   * @return {Promise<void>}
+   */
+  async gotoViewPage(page, row) {
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/orders/shoppingCarts/index.js]:
```suggestion
  async goToViewPage(page, row) {
```

**Review by nesrineabdmouleh — COMMENTED** [review]
Small feedback

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js (hunk 1 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]: Fixed

File: tests/UI/pages/BO/orders/shoppingCarts/index.js (hunk 5 above)
Comment (Progi1984) [reply tests/UI/pages/BO/orders/shoppingCarts/index.js]: Fixed

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js (hunk 4 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]: Fixed

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js (hunk 3 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]: Fixed

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js (hunk 2 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/03_viewCarts.js]: That already exists. Thanks. Fixed.

**Comment by Progi1984** [comment]
@PrestaShop/qa-automation : Fixed feedbacks from @nesrineabdmouleh ;)

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @Progi1984

**Pull request merged**